huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go

```diff
@@ -245,72 +255,113 @@ func buildDissociateResourcesForDisk(rType string, resources []interface{}) []st
 func buildDissociateResources(vType string, resources *schema.Set) ([]string, error) {
 	rType, ok := ResourceType[vType]
 	if !ok {
-		return nil, fmtp.Errorf("Invalid resource type: %s", vType)
+		return nil, fmt.Errorf("invalid resource type: %s", vType)
 	}
-	logp.Printf("[DEBUG] The resource type is %s", rType)
+	log.Printf("[DEBUG] The resource type is %s", rType)
 	switch rType {
 	case ResourceTypeServer:
 		return buildDissociateResourcesForServer(rType, resources.List()), nil
 	case ResourceTypeDisk, ResourceTypeTurbo:
 		return buildDissociateResourcesForDisk(rType, resources.List()), nil
 	default:
-		return nil, fmtp.Errorf("The vault type only support server, disk and turbo.")
+		return nil, fmt.Errorf("the vault type only support server, disk and turbo")
 	}
 }
 
-func buildCBRVaultBilling(d *schema.ResourceData) *vaults.BillingCreate {
+func isPrePaid(d *schema.ResourceData) bool {
+	if val, ok := d.GetOk("charging_mode"); ok && val.(string) == "prePaid" {
+		return true
+	}
+	return false
+}
+
+func buildBillingStructure(d *schema.ResourceData) *vaults.BillingCreate {
 	billing := &vaults.BillingCreate{
 		ObjectType:      d.Get("type").(string),
 		ConsistentLevel: d.Get("consistent_level").(string),
 		ProtectType:     d.Get("protection_type").(string),
 		Size:            d.Get("size").(int),
 	}
 
+	if isPrePaid(d) {
+		autoRenew, _ := strconv.ParseBool(d.Get("auto_renew").(string))
+		autoPay, _ := strconv.ParseBool(d.Get("auto_pay").(string))
```

Comment:
please use common.GetAutoPay

Reply:
Negative! This parameter is type boolean.

Reply:
I mean you can replace `d.Get("auto_pay").(string)` with `common.GetAutoPay(d)`

Reply:
done.